io/src/main/java/org/commonjava/maven/ext/io/JSONIO.java

```diff
@@ -15,50 +15,200 @@
+import com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper;
+import com.fasterxml.jackson.core.util.BufferRecycler;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
 import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.jayway.jsonpath.Configuration;
 import com.jayway.jsonpath.DocumentContext;
 import com.jayway.jsonpath.JsonPath;
 import com.jayway.jsonpath.Option;
+
 import org.commonjava.maven.ext.common.ManipulationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
-import javax.print.Doc;
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 
 @Named
 @Singleton
 public class JSONIO
 {
-    private final Logger logger = LoggerFactory.getLogger( getClass() );
+    private static final Logger logger = LoggerFactory.getLogger( JSONIO.class );
+
+    private File jsonFile;
 
-    private ObjectMapper mapper = new ObjectMapper( );
+    private JsonEncoding encoding;
 
-    private Configuration conf = Configuration.builder().options( Option.ALWAYS_RETURN_LIST ).build();
+    private Charset charset;
 
-    private PrettyPrinter dpp = new MyPrettyPrinter();
+    private String eol;
+
+    public static JsonEncoding detectEncoding(File jsonFile) throws ManipulationException
+    {
+        try ( FileInputStream in = new FileInputStream( jsonFile ) )
+        {
+            byte[] inputBuffer = new byte[4];
+            in.read( inputBuffer );
+            IOContext ctxt = new IOContext( new BufferRecycler(), null, false );
+            ByteSourceJsonBootstrapper strapper = new ByteSourceJsonBootstrapper( ctxt, inputBuffer, 0,
+                    4 );
+            JsonEncoding encoding = strapper.detectEncoding();
+            logger.debug( "Detected JSON encoding {} for file {}", encoding, jsonFile );
+            return encoding;
+        }
+        catch ( IOException e )
+        {
+            logger.error( "Unable to detect charset for file: {}", jsonFile, e );
+            throw new ManipulationException( "Unable to detect charset for file: " + jsonFile, e );
+        }
+    }
 
-    public DocumentContext parseJSON (final File jsonFile) throws ManipulationException
+    public JsonEncoding getEncoding() throws ManipulationException
+    {
+        return this.encoding;
+    }
+
+    public Charset getCharset() throws ManipulationException
+    {
+        return this.charset;
+    }
+
+    public String detectEOL( File jsonFile ) throws ManipulationException
+    {
+        if ( charset == null )
+        {
+            detectEncoding( jsonFile );
+        }
+
+        if ( StandardCharsets.UTF_16BE.equals( charset ) )
```

Comment:
This logic can all be simplified to `System.lineSeparator().getBytes( charset )`. The method is currently expected to return a string and not a byte array.